pkg/kosmosctl/unjoin/unjoin.go

```diff
@@ -123,37 +161,55 @@ func (o *CommandUnJoinOptions) Run(f ctlutil.Factory, cmd *cobra.Command, args [
+	// 2. delete operator
+	err := o.Client.CoreV1().ServiceAccounts(util.DefaultNamespace).Delete(context.TODO(), clusterlinkOperator, metav1.DeleteOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return fmt.Errorf("(operator) kosmosctl unjoin run error, delete serviceaccout failed: %s", err)
 	}
-	err = o.Client.AppsV1().Deployments(util.DefaultNamespace).Delete(context.TODO(), ClusterlinkOperator, metav1.DeleteOptions{})
+	err = o.Client.AppsV1().Deployments(util.DefaultNamespace).Delete(context.TODO(), clusterlinkOperator, metav1.DeleteOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return fmt.Errorf("(operator) kosmosctl unjoin run error, delete deployment failed: %s", err)
 	}
 
-	//delete secret
-	err = o.Client.CoreV1().Secrets(util.DefaultNamespace).Delete(context.TODO(), utils.ControlPanelSecretName, metav1.DeleteOptions{})
+	// 3. delete secret
+	err = o.Client.CoreV1().Secrets(util.DefaultNamespace).Delete(context.TODO(), util.ControlPanelSecretName, metav1.DeleteOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return fmt.Errorf("(secret) kosmosctl unjoin run error, delete secret failed: %s", err)
 	}
 
-	//delete namespace
-	err = o.Client.CoreV1().Namespaces().Delete(context.TODO(), util.DefaultNamespace, metav1.DeleteOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("(namespace) kosmosctl unjoin run error, delete namespace failed: %s", err)
-	}
-
-	//delete rbac
-	err = o.Client.RbacV1().ClusterRoleBindings().Delete(context.TODO(), utils.ExternalIPPoolNamePrefix, metav1.DeleteOptions{})
+	// 4. delete rbac
+	err = o.Client.RbacV1().ClusterRoleBindings().Delete(context.TODO(), util.ExternalIPPoolNamePrefix, metav1.DeleteOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return fmt.Errorf("(rbac) kosmosctl unjoin run error, delete clusterrolebinding failed: %s", err)
 	}
-	err = o.Client.RbacV1().ClusterRoles().Delete(context.TODO(), utils.ExternalIPPoolNamePrefix, metav1.DeleteOptions{})
+	err = o.Client.RbacV1().ClusterRoles().Delete(context.TODO(), util.ExternalIPPoolNamePrefix, metav1.DeleteOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return fmt.Errorf("(rbac) kosmosctl unjoin run error, delete clusterrole failed: %s", err)
 	}
 
+	// 5. delete namespace
+	err = o.Client.CoreV1().Namespaces().Delete(context.TODO(), util.DefaultNamespace, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("(namespace) kosmosctl unjoin run error, delete namespace failed: %s", err)
+	}
+
+	klog.Info("Cluster [" + clusterName + "] was removed.")
+	return nil
+}
+
+func (o *CommandUnJoinOptions) runKnode(knodeName string) error {
+	klog.Info("Start removing knode from kosmos control plane...")
+	for {
+		err := o.DynamicClient.Resource(util.KnodeGVR).Namespace("").Delete(context.TODO(), knodeName, metav1.DeleteOptions{})
```

Comment:
Is namespace "" ok?

Reply:
It has been tested here and the process can proceed normally,  ns is not specified when the knode is created